core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -282,15 +282,17 @@ private[spark] class TaskSchedulerImpl(
     for (i <- 0 until shuffledOffers.size) {
       val execId = shuffledOffers(i).executorId
       val host = shuffledOffers(i).host
+      val resources = shuffledOffers(i).resources
       if (availableCpus(i) >= CPUS_PER_TASK) {
```

Comment:
The resources seems still best effort allocation, the only enforcement to pin task to executor is number of cpus available.